app/src/main/kotlin/cz/covid19cz/erouska/net/FirebaseFunctionsRepository.kt

```diff
@@ -0,0 +1,33 @@
+package cz.covid19cz.erouska.net
+
+import com.google.firebase.functions.ktx.functions
+import com.google.firebase.ktx.Firebase
+import cz.covid19cz.erouska.AppConfig.FIREBASE_REGION
+import cz.covid19cz.erouska.db.SharedPrefsRepository
+import cz.covid19cz.erouska.utils.DeviceInfo
+import cz.covid19cz.erouska.utils.LocaleUtils
+import kotlinx.coroutines.tasks.await
+
+class FirebaseFunctionsRepository(
+    private val deviceInfo: DeviceInfo,
+    private val prefsRepository: SharedPrefsRepository
+) {
+
+    suspend fun registerEhrid() {
+        val data = hashMapOf(
+            "platform" to "android",
+            "platformVersion" to deviceInfo.getAndroidVersion(),
+            "manufacturer" to deviceInfo.getManufacturer(),
+            "model" to deviceInfo.getDeviceName(),
+            "locale" to LocaleUtils.getLocale()
+        )
+        val ehrid = checkNotNull(callFunction("RegisterEhrid", data)["ehrid"])
```

Comment:
Dostávám FirebaseFunctionsException: Response is not valid JSON object. Problém na BE?